tests/models/test_default_evaluator.py

```diff
@@ -329,6 +332,45 @@ def test_svm_classifier_evaluation(svm_model_uri, breast_cancer_dataset):
     }
 
 
+def _evaluate_explainer_with_exceptions(model_uri, dataset):
+    with mlflow.start_run():
+        evaluate(
+            model_uri,
+            dataset._constructor_args["data"],
+            model_type="classifier",
+            targets=dataset._constructor_args["targets"],
+            dataset_name=dataset.name,
+            evaluators="default",
+            evaluator_config={
+                "ignore_exceptions": False,
+            },
+        )
+
```

Comment:
not sure if there is a better, more "pytest" way of doing this

Reply:
What do you want to this in this test?

Reply:
To test an exception, you can use `pytest.raises`.


```
with pytest.raises(<ExceptionType>, match="<error message>"):
    ...
```

Reply:
This code is a helper method, I think the code means we run the test without swallowing exceptions.
And all tests should pass and do not raise exception, if raise exception, the test will fail and we can see the exception stack trace.

Reply:
I think I understand what you want to do.

Reply:
`_evaluate_explainer_with_exceptions` is a helper function that runs evaluation with explainability. It is called for various dataframes by:

- df with string columns: `test_default_explainer_pandas_df_str_cols`
- df with integer columns: `test_default_explainer_pandas_df_num_cols`
- df with long-name columns:`test_default_explainer_pandas_df_longname_cols`

If NaNs are generated, an error will be raised by:

```
if not self.evaluator_config.get("ignore_exceptions", True):
    raise e
```

and so, if the above test runs without raising an exception, that is considered a success and the test passes. We don't need `pytest.raises`, as we aren't trying to purposely raise an exception.

I was more asking whether there is a way to parameterize `_evaluate_explainer_with_exceptions` in a more "pytest" way that doesn't require calling the function three times from three separate places.

Reply:
Ok, `_evaluate_explainer_with_exceptions` sounds like `mlflow.evaluate succeeds even when exceptions occur during evaluation`.